### PR TITLE
Make minicart content more flexible

### PR DESCRIPTION
**Pull request by victorhmp**
#### What is the purpose of this pull request?

- Make the `BaseContent` component render any children blocks passed to it.
- Create new interface `minicart-checkout-button` and use `sticky-layout` to make `minicart.v2` much more flexible.
- Update default blocks implementation.

#### What problem is this solving?

This should enable users to further customize the minicart. Enabling developers to do something like this https://app.clubhouse.io/vtex/story/29613/minicart-v2-should-be-more-flexible .

#### How should this be manually tested?

Go to [Workspace](https://minicartchildren--storecomponents.myvtex.com/), add items into the minicart and see that right above the minicart-summary there's a `flex-layout` block. This `flex-layout` was placed there using:

```json
"minicart.v2": {
    "children": ["minicart-base-content"]
  },
  "minicart-base-content": {
    "blocks": ["minicart-empty-state"],
    "children": ["minicart-product-list", "sticky-layout#minicart-footer"]
  },
  "sticky-layout#minicart-footer": {
    "children": [
      "flex-layout.row#deals",
      "minicart-summary",
      "minicart-checkout-button"
    ]
  },
  "minicart-product-list": {
    "blocks": ["product-list#minicart"]
  },
  "product-list#minicart": {
    "blocks": ["product-list-content-mobile"]
  },
  "minicart-summary": {
    "blocks": ["checkout-summary.compact#minicart"]
  },
(...)
```

#### Screenshots or example usage

Screen Shot 2020-01-22 at 10 00 59

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/BaseContent.tsx [hunk 1]
```diff
@@ -67,6 +67,7 @@ const Content: FC<Props> = ({ finishShoppingButtonLink }) => {
       </div>
       {!isCartEmpty && (
         <div className={minicartFooterClasses}>
+          {children}
```
Comment (klzns) [inline react/BaseContent.tsx]: Make it more flexible verifying if children is passed and delegate the layout to the user.

**Review by klzns — CHANGES_REQUESTED** [review]

**Review by iaronaraujo — CHANGES_REQUESTED** [review]
changelog

**Review by klzns — CHANGES_REQUESTED** [review]
Check if sticky-layout can substitute minicart-footer

**Review by Klynger — APPROVED** [review]

**Review by iaronaraujo — APPROVED** [review]

File: react/ProductList.tsx [hunk 2]
```diff
@@ -41,6 +45,21 @@ const ProductList: FC = () => {
     removeItem({ uniqueId })
   }
 
+  if (renderAsChildren) {
+    return (
+      <div
+        className={`w-100 h-100 overflow-y-auto ${handles.minicartProductListContainer}`}
```
Comment (klzns) [inline react/ProductList.tsx]: Isn't it easier to change the classes if the prop is true instead of creating this whole if?

**Review by klzns — APPROVED** [review]

File: react/ProductList.tsx (hunk 2 above)
Comment (victorhmp) [reply react/ProductList.tsx]: Yep!

**Pull request merged**